components/brave_wallet/browser/zcash/zcash_shield_sync_service.cc

```diff
@@ -103,56 +107,41 @@ void ZCashShieldSyncService::ScheduleWorkOnTask() {
 }
 
 void ZCashShieldSyncService::WorkOnTask() {
-  if (stopped_) {
-    return;
-  }
-
   if (error_) {
+    verify_chain_state_task_.reset();
+    scan_blocks_task_.reset();
+
     if (observer_) {
       observer_->OnSyncError(
-          account_id_,
+          context_.account_id,
           base::NumberToString(GetCode(error_->code)) + ": " + error_->message);
     }
     return;
   }
 
-  if (!chain_tip_block_) {
-    UpdateChainTip();
-    return;
-  }
-
   if (!account_meta_) {
     GetOrCreateAccount();
     return;
   }
 
-  if (!latest_scanned_block_) {
-    VerifyChainState(*account_meta_);
-    return;
-  }
-
-  if (!spendable_notes_) {
-    UpdateSpendableNotes();
-    return;
-  }
-
-  if (observer_) {
-    observer_->OnSyncStatusUpdate(account_id_, current_sync_status_.Clone());
-  }
-
-  if (!downloaded_blocks_ && *latest_scanned_block_ < *chain_tip_block_) {
-    DownloadBlocks();
+  if (!chain_state_verified_) {
+    VerifyChainState();
     return;
+  } else {
+    verify_chain_state_task_.reset();
```

Comment:
It seems `verify_chain_state_task_.reset();` should be done with `chain_state_verified_ = true`

Reply:
It is